.github/workflows/conformance-delegated-ipam.yaml

```diff
@@ -0,0 +1,156 @@
+name: Conformance Delegated IPAM
+
+# TODO: update this to run on a schedule.
+on: workflow_dispatch
+
+jobs:
```

Comment:
This is probably missing some necessary boilerplate -- I saw many other steps in the other pipelines, but wasn't sure what they all meant. Please let me know what else I need to add here.

Reply:
I'd suggest to take a look at the "Conformance Multi Pool IPAM" workflow at [`.github/workflows/conformance-multi-pool.yaml`](https://github.com/cilium/cilium/blob/main/.github/workflows/conformance-multi-pool.yaml) for the necessary boilerplate.

Reply:
Thanks! I copied the steps from conformance-multi-pool.yaml into this workflow. Left it in a separate commit for now so it's clear which steps were copied. I'll squash these down before merging.

Now trying to test this on my fork to make sure things are still working with the boilerplate added.

Reply:
Passing on this run I triggered from my fork: https://github.com/wedaly/cilium/actions/runs/10836785242/job/30071291776

I can see that:
1. cilium install is using delegated IPAM parameters.
2. kind cluster is created with three nodes, none of which are assigned a pod CIDR by controller-manager.
3. nodes become ready 
4. coredns pods are assigned pod IP from 10.244.1.0/24 (the the subnet configured for host-local IPAM plugin on the control plane node).
4. cilium connectivity tests pass, which means pods were assigned IPs on the worker nodes as well.

Reply:
rebased and squashed commits